### ISSUE TITLE
Non-transparent override of HttpOverrides.global inside datadog_tracking_http_client

### ISSUE THREAD

**Issue by paul678** [DataDog/dd-sdk-flutter#424]
**Describe what happened**
While using the `enableHttpTracking()` method we run into a problem that our global `userAgent` and http interceptors where not working. After some digging around we pinpointed the issue the fact that the `datadog_tracking_http_client` changes he `HttpOverrides.global` without ever documenting it.

This means that depending on when you do your app `HttpOverrides.global` change you will end up in one of the following situations:

1. DD initialises **before** app changes the `HttpOverrides.global` value => you will loose all functionality provided by `DatadogTrackingHttpOverrides`
2. DD initialises **after** => you will loose all the custom app functionality.

**Steps to reproduce the issue:**

See above.

**Describe what you expected:**

First step would be to clearly state that the plugin is changing the `HttpOverrides.global` instance so developers expect this. 
Second step would be to provide a way to still support custom `HttpOverrides` alongisde with the `DatadogTrackingHttpOverrides`. Probably by adding a parameter to the `enableHttpTracking()` or something like that.

**Additional context**
 - Datadog SDK version: `datadog_flutter_plugin: 1.3.3` 
 - Versions of any other relevant dependencies: `datadog_tracking_http_client 1.2.1`

**Comment by fuzzybinary** [comment]
Hi @paul678,

Thanks for reporting this and I'm sorry for the oversight here. I'll get this documented as well as provide a way to supply your own overrides object.

Thankfully, the `enableHttpTracking` doesn't do much other than setup the `DatadogTrackingHttpOverrides`, and the overrides class itself is public, so you should be able to work around this until I get the new parameter in place. 

Probably the easiest way to do this is to change your HttpOverrides class to extend `DatadogTrackingHttpOverrides` instead, and don't call `enableTrackingHttpClient` on your configuration object. That should still get you all of the automatic resource tracking and distributed tracing functionality.

Let me know if that works for you or you have any issues.

**Comment by fuzzybinary** [comment]
I've added some documentation both to the [pub site](https://pub.dev/documentation/datadog_tracking_http_client/latest/) and to [Datadog's setup documentation](https://docs.datadoghq.com/real_user_monitoring/flutter/#automatically-track-resources) that make the use of HttpOverrides clearer.  I've also made it so version 1.3.0 of the plugin checks if HttpOverrides is already set (via `HttpOverrides.current`) and uses that to create the client instead.  

There are a few edge cases that this doesn't cover (such as changing HttpOverrides post app initialization) but hopefully the side effects are clearer now for folks that need more complex setups.
